### ISSUE TITLE
VertexBufferWriter writes wrong dataSize

### ISSUE THREAD

**Issue by robintheilade** [MonoGame/MonoGame#6655]
#### What version of MonoGame does the bug occur on:
- MonoGame 3.7.1.189

#### What operating system are you using:
- Windows

#### What MonoGame platform are you using:
- WindowsDX

I'm trying to use models from Blender using the .blend file format but VertexBufferWriter writes the wrong dataSize to the XNB file.

`VertexBufferWriter` calculates the dataSize and writes it to the XNB file. As you can see from the screenshot below the result of the calculation is `22`.
![image](https://user-images.githubusercontent.com/1859029/51865694-5fe4ef80-2347-11e9-9314-d61fa33cdbaf.png)

Here the `VertexBufferReader` attempts to load the data again but a dataSize of `528` is wrong. It should have been `544`.
![image](https://user-images.githubusercontent.com/1859029/51865782-a0dd0400-2347-11e9-85ff-183c88540eda.png)

It might be a good idea to store the actual dataSize rather than calculating it?

Note this issue resembles https://github.com/MonoGame/MonoGame/issues/6237 however I don't believe it is 100% the same.

This is the Blender model I used: [Wall.zip](https://github.com/MonoGame/MonoGame/files/2804946/Wall.zip)

**Comment by Halofreak1990** [comment]
The problem is that your source vertex array is invalid; it's length is not a whole multiple of the VertexStride, which you can see when performing the division in Windows Calculator.

**Comment by robintheilade** [comment]
Hi Tom,

I cannot see your comment on github so I wondered if you deleted again.
However, you are right, the model looks horrible when rendered by XNA if I
make the chance I suggested.

So now my suggestion is that an exception is to thrown if the vertex array
is invalid at build time, so the exception message is displayed in the
content builder output. The issue is that depending on how invalid the
vertex buffer is you will see different exception messages due to the wrong
offsets while reading the XNB file. This makes it very difficult to search
for a solution to the problem and stepping in the MonoGame source code to
figure out what is causing the problem like I did takes a lot of time.

**Comment by Halofreak1990** [comment]
@robintheilade that is correct. I re-read your post and the linked issue, and felt my answer didn't really fit the situation, so I deleted it. It was the first thing that came to mind at the time I wrote it, and I hadn't looked at the linked issue, yet.

I'll quote my post below, as I now realize it indeed had some merit, even if only to point out that the issue seems to lie in or near the VertexBufferContent class.
> The problem is that your source vertex array is invalid; it's length is
> not a whole multiple of the VertexStride, which you can see when performing
> the division in Windows Calculator.

**Comment by robintheilade** [comment]
Placing the following code inside the `ModelProcessor.ProcessMesh` method will display a somewhat more useful error message in the content builder.

```csharp
// throw an error if length of VertexData is not divisible by the length of VertexStride
// if we do not fail here VertexBufferWriter will write a size that VertexBufferReader
// cannot use to recalculate the correct VertexData size and hence not read enough bytes
if (vertexBuffer.VertexData.Length % vertexBuffer.VertexDeclaration.VertexStride > 0)
{
	throw new PipelineException($"VertexData size ({vertexBuffer.VertexData.Length}) must be divisible by the VertexStride size ({vertexBuffer.VertexDeclaration.VertexStride}) in geometry ({geometry.Name ?? mesh.Name}).");
}
```

As seen here:
![image](https://user-images.githubusercontent.com/1859029/52006294-1540b000-24cc-11e9-934d-066f35b3457d.png)

Thoughts?
